lib/SlackBot.js

```diff
@@ -207,19 +207,24 @@ function Slackbot(configuration) {
 
         if (slack_botkit.config && slack_botkit.config.webserver && slack_botkit.config.webserver.static_dir)
             static_dir = slack_botkit.config.webserver.static_dir;
-
-        slack_botkit.config.port = port;
+        
+        if (slack_botkit.config) {
+            slack_botkit.config.port = port;
+            slack_botkit.config.ip = slack_botkit.config.ip || '127.0.0.1';
+        }
+        
+        var ip = slack_botkit.config? slack_botkit.config.ip : '127.0.0.1';
 
         slack_botkit.webserver = express();
         slack_botkit.webserver.use(bodyParser.json());
         slack_botkit.webserver.use(bodyParser.urlencoded({ extended: true }));
         slack_botkit.webserver.use(express.static(static_dir));
 
         var server = slack_botkit.webserver.listen(
-            slack_botkit.config.port,
+            port,
```

Comment:
Why did you change this line to port?

If config exists port is not changed. But, slack_botkit.config.port changes to port. Therefore, there exist a logical error.